tests/cloudconfig/orchestrator/orch_cluster.rb

```diff
@@ -32,28 +32,25 @@ def orch_uri(path)
     return URI("http://#{node.hostname}:19071/orchestrator/v1/#{path}")
   end
 
+  def process_response(response)
+    puts "HTTP response code: #{response.code}"
+    puts "HTTP response body: \n>>>>> #{response.body} <<<<<"
+    response
+  end
+
   def orch_get(path)
     uri = orch_uri(path)
-    req = Net::HTTP::Get.new uri.request_uri
-    Net::HTTP.start(uri.host, uri.port) do |session|
-      response = session.request req
-      puts "HTTP response code: #{response.code}"
-      puts "HTTP response body: \n>>>>> #{response.body} <<<<<"
-      return response
-    end
+    process_response(https_client.get(uri.host, uri.port, uri.path))
   end
 
-  # 'func' is either a Put or Delete, depending on whether the request should
-  # suspend (/suspended with PUT), or resume (/suspended with DELETE).
-  def orch_suspended(host, func)
+  def orch_suspend(host)
     uri = orch_uri("hosts/#{host}/suspended")
-    req = func.new uri.request_uri
-    Net::HTTP.start(uri.host, uri.port) do |session|
-      response = session.request req
-      puts "HTTP response code: #{response.code}"
-      puts "HTTP response body: \n>>>>> #{response.body} <<<<<"
-      return response
-    end
+    process_response(https_client.put(uri.host, uri.port, uri.path, nil))
```

Comment:
Could probably skip explicit `nil` for `headers` argument since it's defaulted (unless `nil` vs `{}` has special meaning?)

Reply:
`nil` is for the body parameter that does not have a default value.

Reply:
My mistake, got my function signatures mixed up 🙂